Source/AwsCommonRuntimeKit/io/HostResolver.swift

```diff
@@ -58,20 +52,25 @@ public final class DefaultHostResolver: HostResolver {
     }
 
     public func resolve(host: String) async throws -> [HostAddress] {
+        let resolverOptions: UnsafeMutablePointer<ResolverOptions> = try! allocator.allocate(capacity: 1)
+        defer {
+            allocator.release(resolverOptions)
+        }
```

Comment:
I will try to refactor this into & syntax, as it is being used in only one function call. Keeping it for now to get this approach reviewed when we need a pointer for the lifetime of a function.